### PR TITLE
Expand/Collapse all changed files

### LINKED ISSUE #23628: Unfold/Unview all files
### Feature Description

I find myself coming back to PRs a lot, or after a code review of a Colleague I need to look at my own PR and when I want to review old files it's annoying to unclick "viewed" one-by-one.

I think there are multiple options of achieving this, I'm unsure which is the best approach.

1. A button that marks all files as unviewed (maybe it can also do mark as viewed for double utility)
2. Just have a button expand/fold all files and leaving the viewed status alone
3. Reset the view status after a PR merge (this only helps in some cases, and might just be confusing, so worst option IMO)
4. ???

I don't see any way that GitHub solves this. I think you also need to manually unfold files there. 
Maybe there is a better solution I'm not seeing yet?

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by sillyguodong**
close #23628 

Now in `...` dropdown, you can expand or collapse all diff files that have loaded.

https://user-images.githubusercontent.com/33891828/227749688-2d406916-3347-49f6-93a5-4092a00e8809.mov

### CODE REVIEW

File: templates/repo/diff/box.tmpl [hunk 1]
```diff
@@ -30,6 +30,12 @@
 					<label for="viewed-files-summary" id="viewed-files-summary-label" class="gt-mr-3 gt-f1" data-text-changed-template="{{.locale.Tr "repo.pulls.viewed_files_label"}}">
 						{{.locale.Tr "repo.pulls.viewed_files_label" .Diff.NumViewedFiles .Diff.NumFiles}}
 					</label>
+					<button id="view-all-btn" class="ui button tiny tooltip basic{{if eq .Diff.NumViewedFiles .Diff.NumFiles}} gt-hidden{{end}}" data-content="View All">
```
Comment (silverwind) [inline templates/repo/diff/box.tmpl]: Need to localize tooltip.

File: web_src/js/features/pull-view-file.js [hunk 2]
```diff
@@ -69,3 +80,7 @@ export function initViewedCheckboxListenerFor() {
     });
   }
 }
+
+export function initViewAndUnviewAllButton() {
+
+}
```
Comment (silverwind) [inline web_src/js/features/pull-view-file.js:89]: Empty function?

**Comment by delvh** [comment]
Hmm… We should also discuss if this really needs to be backend functionality.
I've understood the request rather as a purely frontend side feature:
As I've understood it, it should simply expand/collaps all files, and not reset the `Viewed` state.

**Comment by gempir** [comment]
I agree that not modifying the view state is probably wiser. The Usecase is not that common and not sending a bunch of changed data to the backend reduces load. 

So just a open all/fold all button frontendside would probably be enough.

**Comment by sillyguodong** [comment]
thank you guys for your review and suggestion : )
let me try to implement expand/collapse later.

File: web_src/js/features/pull-view-file.js [hunk 3]
```diff
@@ -69,3 +71,20 @@ export function initViewedCheckboxListenerFor() {
     });
   }
 }
+
+export function initExpandFilesButton() {
```
Comment (lunny) [inline web_src/js/features/pull-view-file.js]: Should we need two functions? Maybe one init function is enough.

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
Hmm I don't think I like those icon choices. How about moving these actions into the "..." menu? I think they are not all that common to warrant separate always-visible buttons, and inside that menu, no icons are needed 😉.

**Comment by sillyguodong** [comment]
> Hmm I don't think I like those icon choices. How about moving these actions into the "..." menu? I think they are not all that common to warrant separate always-visible buttons, and inside that menu, no icons are needed 😉.

done~

File: web_src/js/index.js [hunk 4]
```diff
@@ -201,4 +201,5 @@ $(document).ready(() => {
   initUserAuthWebAuthnRegister();
   initUserSettings();
   initViewedCheckboxListenerFor();
+  initExpandAndCollapseFilesButton();
```
Comment (wxiaoguang) [inline web_src/js/index.js]: Should this "init" appear globally? Is there a parent "init" call for it?

File: templates/repo/diff/options_dropdown.tmpl [hunk 5]
```diff
@@ -12,5 +12,9 @@
 			<a class="item" href="{{$.RepoLink}}/commit/{{PathEscape .Commit.ID.String}}.patch" download="{{ShortSha .Commit.ID.String}}.patch">{{.locale.Tr "repo.diff.download_patch"}}</a>
 			<a class="item" href="{{$.RepoLink}}/commit/{{PathEscape .Commit.ID.String}}.diff" download="{{ShortSha .Commit.ID.String}}.diff">{{.locale.Tr "repo.diff.download_diff"}}</a>
 		{{end}}
+		{{if and .PageIsPullFiles $.SignedUserID (not .IsArchived)}}
```
Comment (lunny) [inline templates/repo/diff/options_dropdown.tmpl]: Why `.IsArchived` needs to be checked?

File: web_src/js/features/pull-view-file.js [hunk 6]
```diff
@@ -1,9 +1,12 @@
 import {setFileFolding} from './file-fold.js';
 
+
```
Comment (silverwind) [inline web_src/js/features/pull-view-file.js]:
```suggestion
```

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Tested, works. Sorry I had to abuse your PR to add one unrelated changes that adjust border-radius to fit more into our theme for this label:

Before:
Screenshot 2023-04-09 at 14 15 27

After:
Screenshot 2023-04-09 at 14 15 15

**Pull request merged**